ElectronClient/app/app.js

```diff
@@ -1148,6 +1149,22 @@ class Application extends BaseApplication {
 		this.lastMenuScreen_ = screen;
 	}
 
+
+
+	disableShortCuts() {
+		for (const itemId of ['copy', 'paste', 'cut', 'selectAll', 'bold', 'italic', 'link', 'code', 'insertDateTime', 'commandStartExternalEditing', 'showLocalSearch']) {
+			const menuItem = Menu.getApplicationMenu().getMenuItemById(`edit:${itemId}`);
+			if (!menuItem) continue;
+			menuItem.enabled = false;
+		}
+	}
+	enableShortCuts() {
+		for (const itemId of ['copy', 'paste', 'cut', 'selectAll', 'bold', 'italic', 'link', 'code', 'insertDateTime', 'commandStartExternalEditing', 'showLocalSearch']) {
+			const menuItem = Menu.getApplicationMenu().getMenuItemById(`edit:${itemId}`);
+			if (!menuItem) continue;
+			menuItem.enabled = true;
+		}
+	}
```

Comment:
No reason to duplicate this code.

Use one function with a boolean argument. e.g. `enableShortcuts(enable = true)`.

Please use `Shortcuts` not `ShortCuts`. Additionally I suggest to use a global Array for `['copy', 'paste', 'cut', 'selectAll', 'bold', 'italic', 'link', 'code', 'insertDateTime', 'commandStartExternalEditing', 'showLocalSearch']` since it is also used in `updateMenuItemStates`. It's never a good idea to duplicate code.

Reply:
@tessus Thanks for the review, I’ll make these changes as soon as possible but please can you explain what you mean by global Array?, how do I do that ?

Reply:
Well, maybe global array is the wrong word for OO code. Define it so that it is accessible from both functions (`enableShortcuts` and `updateMenuItemStates`). We don't want the same definition in two different methods.
Then you can do something like: `for (const itemId of menuitemsArray) { ... }`

Reply:
Okay I get it now, that’s really cool.

Reply:
This block of code is copied verbatim from updateMenuItemStates. We can't have business logic duplicated three times like that, that's a nightmare to maintain. As a matter of fact, doing that you've broken the logic to disable edit buttons in HTML notes.

Reply:
I have refactored and I’ve corrected this

Reply:
Nope, If you have a HTML note open (in edit mode, or split view) the edit menu items are still ensbled.
A HTML note is created via the web clipper.

Reply:
Yes it's still the same issue. There are two problems:

1. You've copied and pasted the logic from [here](https://github.com/laurent22/joplin/pull/2529/files#diff-00b0e649e4e574cdf55197c857ccadefL1160) so now there's two places where we need to describe how items are enabled or disabled. Before you copied that logic twice, now it's better because you've copied it only once, but it's still one too many. You need to modify or refactor the existing logic to implement what you want.

2. As tessus said, the logic for HTML notes is broken because you've duplicated the business logic so now the two functions override each other behaviour. Have a look at the code you've copied, please try to understand what it does, and see how you can re-use it.

Reply:
@tessus thanks, sorry I didn’t test for HTML notes so I didn’t see this, how do I create and open an HTML note

Reply:
From all Laurent has said, it means I’ll have to find a way to solve this issue using just the updateMenuItemsState function

Reply:
I’ll work on it and let you know the updates

Reply:
To create a HTML note, you have to use the clipper and pick `Clip complete page (HTML) (Beta)`. The resulting note will be in the format HTML. You can verify that by clicking on the note properties toolbar icon. Next to the field `Markdown` it will say `HTML`.

Reply:
Alright thanks

Reply:
> From all Laurent has said, it means I’ll have to find a way to solve this issue using just the updateMenuItemsState function

Yes, you will have to change the `updateMenuItemsState` function to fit your needs. Maybe add a second parameter with a default value. So, the current use of the function doesn't change. e.g. `enable = null`. If enable is null, do what it does now. And if it is true/false do something else. This is just an idea. Maybe there's a better way. You will have to figure this out.

Reply:
Thanks for the guidance, this is so helpful!

Reply:
You are welcome. Btw, you should rebase. The directory `app` does not exist anymore.
Just in case you don't know git to well, make a backup of your branch and/or the `app.js` file.
Laurent changed the build process. The BUILD.md was updated accordingly.

Reply:
Thanks, I’ll take action.